src/borrow_tracker/tree_borrows/wildcard.rs

```diff
@@ -0,0 +1,598 @@
+                max(src_state.max_foreign_access, old_exposed_as),
+                // Consider all children.
+                src_state.child_reads,
+                src_state.child_writes,
+                node.children.iter().copied(),
+                wildcard_accesses,
+            );
+        }
+        // We need to propagate the tracking info up the tree, for this we traverse up the parents.
+        // We can skip propagating info to the parent and siblings of a node if its access didn't change.
+        {
+            // The child from which we came.
+            let mut child = id;
+            // This is the `max_local_access()` of the child we came from, before this update...
+            let mut old_child_access = src_old_local_access;
+            // and after this update.
+            let mut new_child_access = src_new_local_access;
+            while let Some(parent_id) = nodes.get(child).unwrap().parent {
+                let parent_node = nodes.get(parent_id).unwrap();
+                let mut entry = wildcard_accesses.entry(parent_id);
+                let parent_state = entry.or_insert(Default::default());
+
+                let old_parent_state = parent_state.clone();
+                use WildcardAccessLevel::*;
+                // Updating this node's tracking state for its children.
+                match (old_child_access, new_child_access) {
+                    (None | Read, Write) => parent_state.child_writes += 1,
+                    (Write, None | Read) => parent_state.child_writes -= 1,
+                    _ => {}
+                }
+                match (old_child_access, new_child_access) {
+                    (None, Read | Write) => parent_state.child_reads += 1,
+                    (Read | Write, None) => parent_state.child_reads -= 1,
+                    _ => {}
+                }
+
+                let old_parent_local_access = old_parent_state.max_local_access();
+                let new_parent_local_access = parent_state.max_local_access();
+
+                {
+                    // We need to update the `max_foreign_access` of `child`'s siblings.
+                    // For this we can reuse the `push_relevant_children` function.
+                    //
+                    // We pass it just the siblings without child itself. Since `child`'s
+                    // `max_local_access()` is foreign to all of its siblings we can pass
+                    // it as part of the foreign access.
+
+                    // it doesnt matter wether we read these from old or new
+                    let parent_access =
+                        max(parent_state.exposed_as, parent_state.max_foreign_access);
+
+                    let sibling_reads =
+                        parent_state.child_reads - if new_child_access >= Read { 1 } else { 0 };
+                    let sibling_writes =
+                        parent_state.child_writes - if new_child_access == Write { 1 } else { 0 };
```

Comment:
```suggestion
                        parent_state.child_writes - if new_child_access >= Write { 1 } else { 0 };
```
to make it more symmetric